### ISSUE TITLE
ensime server crashes on start (ClassNotFoundException: scala.tools.nsc.interactive.CompilerControl)

### ISSUE THREAD

**Issue by ruescasd** [ensime/ensime-sublime#61]
I'm trying to get this working on windows 7, using mingw32. 

When running serverStart.sh I get this output:

/c/Users/David/projects/hubound /c/Users/David/projects/ensime-sublime
/c/Users/David/projects/ensime-sublime
ENSIME_CONFIG=/c/Users/David/projects/hubound/.ensime
Reading ENSIME 0.9.10-SNAPSHOT config from /c/Users/David/projects/hubound/.ensime
  -> Using JDK at c:\\tools\\jdk1.7-x64
  -> Using flags -Xmx1024M
  -> Using cache at C:\\Users\\David\\projects\\hubound\\.ensime_cache
  -> Using scala version 2.11.6
  -> Resolving, log available in C:\\Users\\David\\projects\\hubound\\.ensime_cache/Resolution/sbt.log
Changing to C:\\Users\\David\\projects\\hubound\\.ensime_cache/Resolution
Running sbt saveClasspath
  -> Starting ENSIME
13:02:29.713 INFO  None a.e.s.Slf4jLogger - Slf4jLogger started
13:02:29.724 DEBUG EventStream(akka://ENSIME) a.e.EventStream - logger log1-Slf4jLogger started
13:02:29.727 DEBUG EventStream(akka://ENSIME) a.e.EventStream - Default Loggers started
13:02:29.782 INFO  None o.a.c.v.i.StandardFileSystemManager - Using "C:\Users\David\AppData\Local\Temp\vfs_cache" as temporary files store.
13:02:29.815 DEBUG akka://ENSIME/system/IO-TCP/selectors/$a/0 a.i.TcpListener - Successfully bound to /127.0.0.1:51246
13:02:29.817 INFO  None o.e.s.PortUtil$ - creating portfile C:\Users\David\projects\hubound\.ensime_cache\port
13:02:29.817 INFO  akka://ENSIME/user/ensime-main/tcp-server o.e.s.t.TCPServer - Bound server on port 51246
13:02:30.484 INFO  akka://ENSIME/user/ensime-main o.e.s.ServerActor - Environment:
  OS : Windows 7
  Java : Java HotSpot(TM) 64-Bit Server VM 24.45-b08, Java(TM) SE Runtime Environment 1.7.0_45-b18
  Scala : version 2.11.6
  Ensime : 0.9.10-SNAPSHOT
13:02:30.488 DEBUG akka://ENSIME/system/IO-TCP/selectors/$a/1 a.i.TcpListener - Successfully bound to /127.0.0.1:51247
13:02:30.492 INFO  None o.e.s.PortUtil$ - creating portfile C:\Users\David\projects\hubound\.ensime_cache\http
13:02:30.493 INFO  akka://ENSIME/user/ensime-main o.e.s.ServerActor - ENSIME HTTP on /127.0.0.1
13:02:31.468 WARN  None c.j.b.BoneCPConfig - JDBC username was not set in config!
13:02:31.468 WARN  None c.j.b.BoneCPConfig - JDBC password was not set in config!
13:02:31.676 INFO  None o.e.i.SearchService - removing 0 stale files from the index
13:02:31.691 INFO  None o.e.i.ClassfileWatcher - Setting up new file watchers
13:02:31.717 ERROR ActorSystem(ENSIME) a.a.ActorSystemImpl - Uncaught fatal error from thread [ENSIME-akka.actor.default-dispatcher-4] shutting down ActorSystem [ENSIME]
java.lang.NoClassDefFoundError: com/sun/jdi/VMDisconnectedException
	at org.ensime.core.DebugManager$$anonfun$apply$1.apply(DebugManager.scala:22) ~[core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at org.ensime.core.DebugManager$$anonfun$apply$1.apply(DebugManager.scala:22) ~[core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at akka.actor.TypedCreatorFunctionConsumer.produce(Props.scala:343) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.Props.newActor(Props.scala:252) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.newActor(ActorCell.scala:552) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.create(ActorCell.scala:578) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:456) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:478) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:263) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.dispatch.Mailbox.run(Mailbox.scala:219) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397) [akka-actor_2.11-2.3.12.jar:na]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [scala-library-2.11.6.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [scala-library-2.11.6.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [scala-library-2.11.6.jar:na]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [scala-library-2.11.6.jar:na]
Caused by: java.lang.ClassNotFoundException: com.sun.jdi.VMDisconnectedException
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366) ~[na:1.7.0_45]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355) ~[na:1.7.0_45]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.7.0_45]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354) ~[na:1.7.0_45]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425) ~[na:1.7.0_45]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[na:1.7.0_45]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358) ~[na:1.7.0_45]
	... 15 common frames omitted
13:02:31.718 ERROR ActorSystem(ENSIME) a.a.ActorSystemImpl - Uncaught fatal error from thread [ENSIME-akka.actor.default-dispatcher-7] shutting down ActorSystem [ENSIME]
java.lang.NoClassDefFoundError: scala/tools/nsc/interactive/CompilerControl
	at java.lang.ClassLoader.defineClass1(Native Method) ~[na:1.7.0_45]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800) ~[na:1.7.0_45]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142) ~[na:1.7.0_45]
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449) ~[na:1.7.0_45]
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71) ~[na:1.7.0_45]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361) ~[na:1.7.0_45]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355) ~[na:1.7.0_45]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.7.0_45]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354) ~[na:1.7.0_45]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425) ~[na:1.7.0_45]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[na:1.7.0_45]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358) ~[na:1.7.0_45]
	at org.ensime.core.Analyzer$$anonfun$apply$1.apply(Analyzer.scala:312) ~[core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at org.ensime.core.Analyzer$$anonfun$apply$1.apply(Analyzer.scala:312) ~[core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at akka.actor.TypedCreatorFunctionConsumer.produce(Props.scala:343) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.Props.newActor(Props.scala:252) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.newActor(ActorCell.scala:552) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.create(ActorCell.scala:578) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:456) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:478) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:263) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.dispatch.Mailbox.run(Mailbox.scala:219) ~[akka-actor_2.11-2.3.12.jar:na]
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397) [akka-actor_2.11-2.3.12.jar:na]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [scala-library-2.11.6.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [scala-library-2.11.6.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [scala-library-2.11.6.jar:na]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [scala-library-2.11.6.jar:na]
Caused by: java.lang.ClassNotFoundException: scala.tools.nsc.interactive.CompilerControl
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366) ~[na:1.7.0_45]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355) ~[na:1.7.0_45]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.7.0_45]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354) ~[na:1.7.0_45]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425) ~[na:1.7.0_45]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[na:1.7.0_45]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358) ~[na:1.7.0_45]
	... 27 common frames omitted
13:02:31.723 DEBUG akka://ENSIME/system/IO-TCP/selectors/$a/1 a.i.TcpListener - Unbinding endpoint /127.0.0.1:51247
13:02:31.724 DEBUG akka://ENSIME/system/IO-TCP/selectors/$a/1 a.i.TcpListener - Unbound endpoint /127.0.0.1:51247, stopping listener
13:02:31.725 INFO  akka://ENSIME/system/IO-TCP-STREAM/server-1-%2F127.0.0.1%3A0 a.a.LocalActorRef - Message [akka.io.Tcp$Unbound$] from Actor[akka://ENSIME/system/IO-TCP/selectors/$a/1#-63215872] to Actor[akka://ENSIME/system/IO-TCP-STREAM/server-1-%2F127.0.0.1%3A0#-852894119] was not delivered. [1] dead letters encountered. This logging can be tur
13:02:31.729 DEBUG akka://ENSIME/system/IO-TCP/selectors/$a/0 a.i.SelectionHandler - Monitored actor [Actor[akka://ENSIME/user/ensime-main/tcp-server#-1543673420]] terminated
13:02:31.730 DEBUG akka://ENSIME/system/IO-TCP/selectors/$a/0 a.i.TcpListener - Closing serverSocketChannel after being stopped
13:02:31.778 DEBUG None o.e.i.SearchService - indexing file:///C:/Users/David/.ivy2/cache/org.apache.commons/commons-math3/jars/commons-math3-3.2.jar
13:02:31.778 DEBUG None o.e.i.SearchService - indexing file:///c:/tools/jdk1.7-x64/jre/lib/rt.jar
13:02:31.778 DEBUG None o.e.i.SearchService - indexing file:///C:/Users/David/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.6.jar
13:02:31.785 INFO  None c.j.b.BoneCP - Shutting down connection pool...
13:02:31.786 INFO  None c.j.b.BoneCP - Connection pool has been shutdown.
13:02:31.835 ERROR None o.e.i.SearchService - Failed to index file:///C:/Users/David/.ivy2/cache/org.apache.commons/commons-math3/jars/commons-math3-3.2.jar
org.apache.commons.vfs2.FileSystemException: Could not find files in "jar:file:///C:/Users/David/.ivy2/cache/org.apache.commons/commons-math3/jars/commons-math3-3.2.jar!/".
	at org.apache.commons.vfs2.provider.AbstractFileObject.findFiles(AbstractFileObject.java:1700) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.findFiles(AbstractFileObject.java:1198) ~[commons-vfs2-2.0.jar:2.0]
	at org.ensime.indexer.SearchService.org$ensime$indexer$SearchService$$scan$1(SearchService.scala:58) ~[core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at org.ensime.indexer.SearchService$$anonfun$14$$anonfun$apply$6$$anonfun$apply$3.apply$mcV$sp(SearchService.scala:112) [core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at org.ensime.indexer.SearchService$$anonfun$14$$anonfun$apply$6$$anonfun$apply$3.apply(SearchService.scala:109) [core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at org.ensime.indexer.SearchService$$anonfun$14$$anonfun$apply$6$$anonfun$apply$3.apply(SearchService.scala:109) [core_2.11-0.9.10-SNAPSHOT.jar:0.9.10-SNAPSHOT]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) [scala-library-2.11.6.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) [scala-library-2.11.6.jar:na]
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40) [akka-actor_2.11-2.3.12.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_45]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_45]
	at java.lang.Thread.run(Thread.java:744) [na:1.7.0_45]
Caused by: org.apache.commons.vfs2.FileSystemException: URI "jar:file:///C:/Users/David/.ivy2/cache/org.apache.commons/commons-math3/jars/commons-math3-3.2.jar!//org/apache/commons/math3/stat" is not an absolute file name.
	at org.apache.commons.vfs2.provider.local.GenericFileNameParser.extractRootPrefix(GenericFileNameParser.java:56) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.local.LocalFileNameParser.parseUri(LocalFileNameParser.java:80) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.url.UrlFileNameParser.parseUri(UrlFileNameParser.java:65) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileProvider.parseUri(AbstractFileProvider.java:188) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.impl.DefaultFileSystemManager.resolveName(DefaultFileSystemManager.java:822) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.getChildren(AbstractFileObject.java:710) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.traverse(AbstractFileObject.java:1724) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.traverse(AbstractFileObject.java:1729) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.traverse(AbstractFileObject.java:1729) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.traverse(AbstractFileObject.java:1729) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.traverse(AbstractFileObject.java:1729) ~[commons-vfs2-2.0.jar:2.0]
	at org.apache.commons.vfs2.provider.AbstractFileObject.findFiles(AbstractFileObject.java:1695) ~[commons-vfs2-2.0.jar:2.0]
	... 11 common frames omitted
13:02:31.836 DEBUG EventStream a.e.EventStream - shutting down: StandardOutLogger started

**Comment by rorygraves** [comment]
You have actually gone too far down the stacktrace the error is this one 
```[ENSIME-akka.actor.default-dispatcher-4] shutting down ActorSystem [ENSIME]
java.lang.NoClassDefFoundError: com/sun/jdi/VMDisconnectedException``` which implies you have a JRE set as your JAVA_HOME rather than a JDK - update your path to a JDK and regenerate your ensime file and you should be good.

**Comment by ruescasd** [comment]
As you can see from this line in the output i pasted:

 Using JDK at c:\tools\jdk1.7-x64

It's using a jdk (just double checked that is a jdk), and JAVA_HOME is also pointing to that jdk, so that can't be the answer. Thanks, Rory

**Comment by rorygraves** [comment]
@ruescasd Whoops - missed it (sorry).
Strange, can you check where in you jdk the java debug interface classes (jdi) lives.  And that they end up on the Resolution classpath (```.ensime_cache/Resolution/classpath```).
Not sure quite whats going on there ;)

**Comment by ruescasd** [comment]
Here's what worked for me, seems like classpath separator issue

==================================================
diff --git a/serverStart.sh b/serverStart.sh
index 0457f3d..adfeb44 100755
--- a/serverStart.sh
+++ b/serverStart.sh
@@ -12,6 +12,7 @@
 # [1] https://github.com/paulp/sbt-extras/blob/master/sbt

 ENSIME_VERSION="0.9.10-SNAPSHOT"
+CLASSPATH_SEPARATOR=":"

 # borrowed from http://stackoverflow.com/questions/3466166/how-to-check-if-running-in-cygwin-mac-or-linux
 if [ "$(uname)" == "Darwin" ]; then # Mac
@@ -24,6 +25,7 @@ elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW32_NT" ]; then
   ENSIME_ABS_PATH=`pwd`
   popd
   ENSIME_CONFIG="${ENSIME_ABS_PATH}/.ensime"
+  CLASSPATH_SEPARATOR=";"
 fi

 echo "ENSIME_CONFIG=$ENSIME_CONFIG"
@@ -109,7 +111,7 @@ cd "$RESOLUTION_DIR"
 echo "Running sbt saveClasspath"
 sbt update saveClasspath > "$CLASSPATH_LOG"

-CLASSPATH="$JDK_HOME/lib/tools.jar:`cat $CLASSPATH_FILE`"
+CLASSPATH="$JDK_HOME/lib/tools.jar$CLASSPATH_SEPARATOR`cat $CLASSPATH_FILE`"

 echo "  -> Starting ENSIME"
 cd "$ENSIME_CACHE"

**Comment by rorygraves** [comment]
@ruescasd Good catch.  Want to submit a PR with that fix in?

**Comment by ruescasd** [comment]
https://github.com/ensime/ensime-sublime/pull/62

**Comment by rorygraves** [comment]
Closed thanks to #62
